bot/exts/utilities/challenges.py

```diff
@@ -0,0 +1,323 @@
+import logging
+from random import choice
+from typing import Union
+
+from bs4 import BeautifulSoup
+from discord import Color, Embed, Interaction, SelectOption, ui
+from discord.ext import commands
+
+from bot.bot import Bot
+from bot.constants import Colours, Emojis, NEGATIVE_REPLIES
+
+logger = logging.getLogger(__name__)
+API_ROOT = "https://www.codewars.com/api/v1/code-challenges/{kata_id}"
+# Map difficulty for the kata to color we want to display in the embed.
+# These colors are representative of the colors that each 'kyu' level represents on codewars.com
+MAPPING_OF_KYU = {
+    8: (221, 219, 218), 7: (221, 219, 218), 6: (236, 182, 19), 5: (236, 182, 19),
+    4: (60, 126, 187), 3: (60, 126, 187), 2: (134, 108, 199), 1: (134, 108, 199),
+}
```

Comment:
Technically this could just be a list where we do the lookup by index. That said, then it will start at 0 not 1.

Not sure what's best

Reply:
Hm, I honestly think that the dictionary is more fitting since it is a mapping